lib/facterdb.rb

```diff
@@ -150,4 +150,20 @@ def self.get_facts(filter = nil, cache = true)
     end
     result
   end
+
+  # @param results [Array[Hash]] The results from a get_facts call
+  # @param filter [Object] The filter to convert to jgrep string
+  # @return [Array[Hash[Symbol, Any]]] Array of hashes of facts
+  def self.filter_results(results, filter)
+    # TODO: it's really inefficient to do to_s and to_sym all the time
```

Comment:
I think we should rebase this for the facterdb 3 release when we switch to strings by default